File: sdk/typescript/src/security_assurance.ts
/**
 * TypeScript SDK client for Security Assurance platform.
 */

export interface SecurityAssetPayload {
  name: string;
  asset_type?: string;
  criticality?: string;
  location?: string;
  owner?: string;
  metadata?: Record<string, any>;
}

export class SecurityAssuranceClient {
  private baseUrl: string;
  private apiKey: string;

  constructor(baseUrl: string, apiKey: string) {
    this.baseUrl = baseUrl;
    this.apiKey = apiKey;
  }

  async registerAsset(payload: SecurityAssetPayload, tenantId: string = 'default_tenant'): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/security/assets`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${this.apiKey}`,
        'x-tenant-id': tenantId,
      },
      body: JSON.stringify(payload),
    });
    return res.json();
  }

  async evaluatePosture(tenantId: string = 'default_tenant'): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/security/posture`, {
      method: 'GET',
      headers: {
        'Authorization': `Bearer ${this.apiKey}`,
        'x-tenant-id': tenantId,
      },
    });
    return res.json();
  }

  async evaluateAssurance(tenantId: string = 'default_tenant'): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/security/assurance`, {
      method: 'GET',
      headers: {
        'Authorization': `Bearer ${this.apiKey}`,
        'x-tenant-id': tenantId,
      },
    });
    return res.json();
  }
}

File: sdk/go/security_assurance.go
package llmengine

import (
	"bytes"
	"context"
	"encoding/json"
	"net/http"
)

type SecurityAssuranceService struct {
	client *Client
}

type SecurityAssetPayload struct {
	Name        string                 `json:"name"`
	AssetType   string                 `json:"asset_type"`
	Criticality string                 `json:"criticality"`
	Location    string                 `json:"location"`
	Owner       string                 `json:"owner"`
	Metadata    map[string]interface{} `json:"metadata"`
}

func (s *SecurityAssuranceService) RegisterAsset(ctx context.Context, tenantID string, payload SecurityAssetPayload) (map[string]interface{}, error) {
	buf, err := json.Marshal(payload)
	if err != nil {
		return nil, err
	}
	req, err := http.NewRequestWithContext(ctx, "POST", s.client.baseURL+"/v1/security/assets", bytes.NewReader(buf))
	if err != nil {
		return nil, err
	}
	req.Header.Set("Content-Type", "application/json")
	req.Header.Set("x-tenant-id", tenantID)

	res, err := s.client.hc.Do(req)
	if err != nil {
		return nil, err
	}
	defer res.Body.Close()

	var result map[string]interface{}
	err = json.NewDecoder(res.Body).Decode(&result)
	return result, err
}

func (s *SecurityAssuranceService) EvaluatePosture(ctx context.Context, tenantID string) (map[string]interface{}, error) {
	req, err := http.NewRequestWithContext(ctx, "GET", s.client.baseURL+"/v1/security/posture", nil)
	if err != nil {
		return nil, err
	}
	req.Header.Set("x-tenant-id", tenantID)

	res, err := s.client.hc.Do(req)
	if err != nil {
		return nil, err
	}
	defer res.Body.Close()

	var result map[string]interface{}
	err = json.NewDecoder(res.Body).Decode(&result)
	return result, err
}
